plugins/notifications-backend/src/service/router.ts

```diff
@@ -1,62 +1,66 @@
 import { errorHandler } from '@backstage/backend-common';
-
 import express from 'express';
 import Router from 'express-promise-router';
-import { Knex } from 'knex';
+import { CatalogClient } from '@backstage/catalog-client';
 import { Logger } from 'winston';
+import { Config } from '@backstage/config';
+
+import { initDB } from './db';
+import { createNotification, getNotifications, getNotificationsCount } from './handlers';
 
-export interface RouterOptions {
+interface RouterOptions {
   logger: Logger;
-  dbClient?: Knex<any, any[]>;
+  dbConfig: Config;
+  catalogClient: CatalogClient;
 }
 
 export async function createRouter(
   options: RouterOptions,
 ): Promise<express.Router> {
-  const { logger, dbClient } = options;
+  const { logger, dbConfig, catalogClient } = options;
 
   const router = Router();
   router.use(express.json());
 
-  if (!dbClient) {
-    logger.error('Missing dbClient');
-    return router;
+  if (!dbConfig) {
+    logger.error('Missing dbConfig');
+    throw new Error('Missing database config');
   }
 
-  router.get('/health', (_, response) => {
-    logger.info('PONG!');
+  // create DB client an tables
+  const dbClient = await initDB(dbConfig);
+
+  // rest endpoints/operations
+  router.get('/health', async (_, response) => {
     response.json({ status: 'ok' });
   });
 
-  router.get('/notifications/count', async (_, response) => {
-    let msgcount = -1;
-    await dbClient('messages')
-      .count('* as CNT')
-      .then(count => {
-        msgcount = (count[0]?.CNT as unknown as number) || -1;
-      });
-    response.json({ status: 'ok', count: msgcount });
+  router.get('/notifications/count', async (request, response) => {
+    getNotificationsCount(dbClient, request.query)
+    .then( result => response.json(result) );
   });
+  
+  router.get('/notifications', async (request, response) => {
+    const {pageSize, pageNumber} = request.query;
 
-  router.post('/notifications', async (request, response) => {
-    let msgid = -1;
-    await dbClient('messages')
-      .insert(request.body)
-      .returning('id')
-      .then(id => {
-        // We should harden the type
-        msgid = id as unknown as number;
-      });
-    response.json({ status: 'ok', msgid: msgid });
+    if ( typeof pageSize !== 'string' || typeof pageNumber !== 'string' ) {
+      throw new Error('either pageSize or pageNumber query string parameters are missing/invalid');
+    }
+
+    const pageSizeNum = Number.parseInt(pageSize);
+    const pageNumberNum = Number.parseInt(pageNumber);
+
+    if ( Number.isNaN(pageSizeNum) || Number.isNaN(pageNumberNum) ) {
+      throw new Error('either pageSize or pageNumber is not a number');
```

Comment:
I think this check is good-enough (I feel like we do not need the `typeof` check above)

Reply:
I need 2 checks. why? the parameters can be undefined/string/string[]. HTTP allows multiple query parameters with same name. So my first check verifies it is not undefined nor array of strings.
second check is after parsing into int.
compliation fails if I pass pageSize to parseInt without the first check